File: scala-3.6/template-renderer/src/main/scala/com/rbleggi/templaterenderer/renderers/PDFRenderer.scala
package com.rbleggi.templaterenderer.renderers

import com.rbleggi.templaterenderer.core.TemplateRenderer
import com.itextpdf.kernel.pdf.{PdfDocument, PdfWriter}
import com.itextpdf.layout.Document
import com.itextpdf.layout.element.Paragraph
import java.io.ByteArrayOutputStream

class PDFRenderer extends TemplateRenderer {
  def render(data: Map[String, String]): Array[Byte] =
    println("Generating PDF Template...")
    val outputStream = new ByteArrayOutputStream()
    val document = new Document(new PdfDocument(new PdfWriter(outputStream)))

    document.add(new Paragraph(s"Title: ${data("title")}"))
    document.add(new Paragraph(s"Content: ${data("content")}"))

    document.close()
    println("PDF Template Generated Successfully!")
    outputStream.toByteArray
}

File: kotlin/template-renderer/src/main/kotlin/com/rbleggi/templaterenderer/service/renderer/PdfRenderer.kt
package com.rbleggi.templaterenderer.service.renderer

import com.itextpdf.kernel.pdf.PdfDocument
import com.itextpdf.kernel.pdf.PdfWriter
import com.itextpdf.layout.Document as PdfLayoutDocument
import com.itextpdf.layout.element.Paragraph as PdfParagraph
import java.io.ByteArrayOutputStream

class PdfRenderer : TemplateRenderer() {
    override fun renderTitle(title: String): String = title

    override fun renderContent(content: String): String = content

    override fun assemble(title: String, content: String): ByteArray {
        val output = ByteArrayOutputStream()
        val pdf = PdfLayoutDocument(PdfDocument(PdfWriter(output)))
        pdf.add(PdfParagraph(title))
        pdf.add(PdfParagraph(content))
        pdf.close()
        return output.toByteArray()
    }
}
